src/generate/peripheral.rs

```diff
@@ -48,14 +48,15 @@ pub fn render(
     // Insert the peripheral structure
     out.push(quote! {
         #[doc = #description]
-        pub struct #name_pc { _marker: PhantomData<*const ()> }
+        pub struct #name_pc { _marker: PhantomData<()> }
 
         unsafe impl Send for #name_pc {}
 
         impl #name_pc {
             /// Returns a pointer to the register block
-            pub fn ptr() -> *const #base::RegisterBlock {
-                #address as *const _
+            #[inline(always)]
+            pub const fn ptr() -> *const #base::RegisterBlock {
+                #address as _
```

Comment:
incorrect cast here (`*i32` instead of `*<width>`)

Reply:
That is very weird. Why could a elided type result in an incompatible pointer?

Reply:
Most likely the issue is not this change but the `PhantomMarker` change above. Anyway let's back that out.